build/targets/KoreBuild.SolutionBuild.targets

```diff
@@ -41,27 +45,60 @@ as they are subject to change.
 -->
 
   <Target Name="CleanArtifacts">
-    <!-- intentionally BuildDir, not ArtifactDir -->
-    <RemoveDir Directories="$(BuildDir)" />
+    <ItemGroup>
+      <_FilesToDelete Include="$(ArtifactsDir)**\*" Exclude="$(ArtifactsDir)msbuild\**\*" />
+    </ItemGroup>
+    <Delete Files="@(_FilesToDelete)" />
+  </Target>
+
+  <Target Name="_PrepareOutputPaths">
+    <MakeDir Directories="$(ArtifactsDir);$(BuildDir)" />
+  </Target>
+
+  <Target Name="_ResolveCommitHash" Condition="'$(CommitHash)'==''">
+    <PropertyGroup>
+      <CommitHash Condition="'$(APPVEYOR_REPO_COMMIT)' != ''">$(APPVEYOR_REPO_COMMIT)</CommitHash>
+      <CommitHash Condition="'$(TRAVIS_COMMIT)' != ''">$(TRAVIS_COMMIT)</CommitHash>
```

Comment:
There is something similiar: https://confluence.jetbrains.com/display/TCD10/Predefined+Build+Parameters, but it requires knowing the 'VCS Id' as TeamCity allows you to attach multiple VCS roots per build config. To make this work correctly, we'd have to update our CI settings to rely on the TeamCity agent to do the git clone instead of having our build.cmd script execute git clone.